exonum-java-proofs/src/main/java/com/exonum/binding/storage/proofs/map/flat/UncheckedFlatMapProof.java

```diff
@@ -27,53 +27,49 @@
 
   private final List<MapProofEntry> proofList = new ArrayList<>();
 
-  private final List<MapProofEntry> branches;
+  private final List<MapProofEntry> proof;
 
-  private final List<MapProofEntryLeaf> leaves;
+  private final List<MapEntry> leaves;
 
-  private final List<MapProofAbsentEntryLeaf> absentLeaves;
+  private final List<byte[]> absentLeaves;
 
   UncheckedFlatMapProof(
-      List<MapProofEntry> branches,
-      List<MapProofEntryLeaf> leaves,
-      List<MapProofAbsentEntryLeaf> absentLeaves) {
-    this.branches = branches;
+      List<MapProofEntry> proof,
+      List<MapEntry> leaves,
+      List<byte[]> absentLeaves) {
+    this.proof = proof;
     this.leaves = leaves;
     this.absentLeaves = absentLeaves;
-    mergeLeavesWithBranches();
   }
 
   @SuppressWarnings("unused") // Native API
   static UncheckedFlatMapProof fromUnsorted(
       MapProofEntry[] proofList,
-      MapProofEntryLeaf[] leaves,
-      MapProofAbsentEntryLeaf[] absentLeaves) {
-    List<MapProofEntry> branchesList = Arrays.asList(proofList);
-    List<MapProofEntryLeaf> leavesList = Arrays.asList(leaves);
-    List<MapProofAbsentEntryLeaf> absentLeavesList = Arrays.asList(absentLeaves);
-    return new UncheckedFlatMapProof(branchesList, leavesList, absentLeavesList);
-  }
-
-  @Override
-  public List<MapProofEntry> getProofList() {
-    return new ArrayList<>(proofList);
+      MapEntry[] leaves,
+      byte[] absentLeaves) {
```

Comment:
I'd avoid using the same name (= leaf) for different types: we can say that both MapEntry and missing keys _correspond_ to leaf nodes in the map tree, but we can't say that they _are_ leaf nodes (as a leaf node is a 34-byte _database_ key + value). Therefore I'd suggest using `entries` and `missingKeys`.